studio/components/interfaces/Database/Backups/BackupItem.tsx

```diff
@@ -1,29 +1,55 @@
 import { PermissionAction } from '@supabase/shared-types/out/constants'
 import dayjs from 'dayjs'
-import { Badge, Button } from 'ui'
+import { Badge, Button, IconDownload } from 'ui'
 
+import { useBackupDownloadMutation } from 'data/database/backup-download-mutation'
 import { DatabaseBackup } from 'data/database/backups-query'
 import { useCheckPermissions } from 'hooks'
 
 interface BackupItemProps {
   index: number
+  projectRef: string
   backup: DatabaseBackup
   onSelectBackup: () => void
 }
 
-const BackupItem = ({ index, backup, onSelectBackup }: BackupItemProps) => {
+const BackupItem = ({ index, backup, projectRef, onSelectBackup }: BackupItemProps) => {
   const canTriggerScheduledBackups = useCheckPermissions(
     PermissionAction.INFRA_EXECUTE,
     'queue_job.restore.prepare'
   )
 
+  const { mutate: downloadBackup, isLoading: isDownloading } = useBackupDownloadMutation({
+    onSuccess: (res) => {
+      const { fileUrl } = res
+
+      // Trigger browser download by create,trigger and remove tempLink
+      const tempLink = document.createElement('a')
+      tempLink.href = fileUrl
+      document.body.appendChild(tempLink)
+      tempLink.click()
+      document.body.removeChild(tempLink)
+    },
+  })
+
   const generateSideButtons = (backup: any) => {
     if (backup.status === 'COMPLETED')
       return (
         <div className="flex space-x-4">
           <Button type="default" disabled={!canTriggerScheduledBackups} onClick={onSelectBackup}>
             Restore
           </Button>
+          {!backup.isPhysicalBackup && (
```

Comment:
This is a new condition as the ability to download a backup is only for logical backups.